Assignment 2.Rmd

```diff
@@ -87,14 +95,72 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
-
-
+```{r}
+EDGES.st <- EDGE %>% group_by(to) %>% summarize(sum(count))
+names(EDGES.st) <- c("id","count")
+EDGES.tc <- EDGE %>% group_by(from) %>% summarize(sum(count))
+names(EDGES.tc) <- c("id","count")
+total <- bind_rows(EDGES.st, EDGES.tc)
+VERTEX2 <- full_join(VERTEX, total, by="id")
+g2 <- graph.data.frame(EDGE, directed=TRUE,vertices=VERTEX2)
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=VERTEX2$gender)
+```
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+#install.packages("tidyr")
+library(dplyr)
+library(tidyr)
+hudk4050_classes<-read.csv("/Users/carolinejin/Desktop/Columbia/Data Mining/assignment2/hudk4050-classes.csv")
+h1 <- hudk4050_classes%>% unite(Name, First.Name, Last.Name, sep= " ")
+h2 <- gather(h1,key="class number", value="class",2:7)
+
+h2 <- na.omit(h2)
+h2 = select(h2, -c("class number"))
+h2$class <- gsub("[[:space:]]", "", h2$class)
+h2 <- h2%>%filter(Name!='ZIMO CHEN')
+h2 <- h2%>%filter(class!="HUDK4050")
+h2$count=1
+
+length.h2<-length(h2[,1])
+h2.new<-h2
+h2.new
+n=0
+for (i in 1:length.h2) {
+  
+if (h2[i,2]=="") { 
+   h2.new<-h2.new[-i+n,]
+   n=n+1
+
+}
+}
```

Comment:
Interesting way to wrangling data. Not sure if you are using the loop to remove blank rows. It would be helpful to add some notations and comments using # to explain the steps.